addons/mail/models/mail_activity_type.py

```diff
@@ -125,3 +125,51 @@ def _inverse_triggered_next_type_id(self):
                 activity_type.chaining_type = 'trigger'
             else:
                 activity_type.chaining_type = 'suggest'
+
+    def write(self, values):
+        """ Protect some master types against deletiong or modification when they
+        are used as default in apps, in critical flows, plans, ... """
+        if values.get('res_model'):
```

Comment:
What about writing "False" on an activity type that requires a specific model? I think it would work and it shouldn't, as we skip your new code, unless I missed something.